apps/editing-toolkit/editing-toolkit-plugin/tutorials/test/tutorials-test.php

```diff
@@ -0,0 +1,112 @@
+<?php
+// phpcs:ignoreFile
+
+use PHPUnit\Framework\TestCase;
+
+require_once dirname( __FILE__, 2 ) . '/tutorials.php';
+
+class WPCom_Tutorials_Test extends TestCase {
+
+	public function tearDown() {
+		// cleanse after each test
+		foreach ( wpcom_get_all_tutorials() as $id ) {
+			wpcom_tutorials()->unregister( $id );
+		}
+		// delete status storage
+		delete_option( WPCom_Tutorials::OPTION_NAME );
+	}
+
+	public function test_registered_tutorial_is_returned() {
+		wpcom_register_tutorial(
+			'foo',
+			array(
+				'title' => 'Foo',
+				'tasks' => $this->get_tasks(),
+			)
+		);
+
+		$return_tutorial = wpcom_get_tutorial( 'foo' );
+
+		$expected_tutorial = array(
+			'id'    => 'foo',
+			'title' => 'Foo',
+			'tasks' => $this->get_tasks_with_status(),
+		);
+
+		$this->assertEquals( $return_tutorial, $expected_tutorial );
+	}
+
+	public function test_all_registered_ids_are_returned() {
+		$this->assertEquals( wpcom_get_all_tutorials(), array() );
+
+		wpcom_register_tutorial(
+			'foo',
+			array(
+				'title' => 'Foo',
+				'tasks' => $this->get_tasks(),
+			)
+		);
+		wpcom_register_tutorial(
+			'bar',
+			array(
+				'title' => 'Bar',
+				'tasks' => $this->get_tasks(),
+			)
+		);
+		wpcom_register_tutorial(
+			'baz',
+			array(
+				'title' => 'Baz',
+				'tasks' => $this->get_tasks(),
+			)
+		);
+
+		$this->assertEquals( wpcom_get_all_tutorials(), array( 'foo', 'bar', 'baz' ) );
+	}
+
+	public function test_tutorial_with_malformed_options_fails() {
+		// empty option object
+		$foo1 = wpcom_register_tutorial( 'foo', array() );
+		$this->assertEmpty( wpcom_get_all_tutorials() );
+		$this->assertFalse( $foo1 );
+
+		// no title
+		$foo2 = wpcom_register_tutorial( 'foo', array( 'tasks' => $this->get_tasks() ) );
+		$this->assertEmpty( wpcom_get_all_tutorials() );
+		$this->assertFalse( $foo2 );
+
+		// no steps
+		$foo3 = wpcom_register_tutorial( 'foo', array( 'title' => 'Foo' ) );
+		$this->assertEmpty( wpcom_get_all_tutorials() );
+		$this->assertFalse( $foo3 );
+
+		// empty steps
+		$foo4 = wpcom_register_tutorial(
+			'foo',
+			array(
+				'title' => 'Foo',
+				'steps' => array(),
+			)
+		);
+		$this->assertEmpty( wpcom_get_all_tutorials() );
+		$this->assertFalse( $foo4 );
+	}
+
```

Comment:
do you think we should have any more tests for setting / retrieving saved statuses?

Reply:
Yup definitely, I'd just hit EOD and wanted to get some stuff up

Reply:
Awesome, thanks for that! I enjoyed being able to take a look this morning, very exciting stuff!

Reply:
Could we also test a nonexistent task `wpcom_tutorial_mark_task` with a nonexistent task ( not `completed`, `skipped`, or `pending` ), a nonexistent task id, and a nonexistent tutorial id?

Reply:
I'm also curious if we handle errors if, for some reason, the call to `persist_statuses` fails

Reply:
There's no particular error-handling code, no. If those DB writes/reads fail, we've got bigger problems.